### ISSUE TITLE
Inflexible score function of CheckpointSaver

### ISSUE THREAD

**Issue by sandylaker** [Project-MONAI/MONAI#2467]
https://github.com/Project-MONAI/MONAI/blob/9f9b63a7dc56d8b31d4fc8d34aae77865ba95947/monai/handlers/checkpoint_saver.py#L148-L160

Currently the `CheckpointSaver` saves the check point that has the largest score value. However, this is not desirable when a lower score indicates better performance, e.g. validation loss. We can either allow the user to pass a custom score function (as in the ignite.handlers), or add a flag to control returning `round(engine.state.metrics[metric_name], 4)` or ` - round(engine.state.metrics[metric_name], 4)`

**Comment by Nic-Ma** [comment]
Hi @sandylaker , thanks for raising this question, I think it's a question for ignite `Checkpoint` handler.
@vfdev-5 , could you please help share some comments? I think you have encountered this question.
Maybe adding a ”-“ before the metrics value if the smaller is better?

Thanks.

**Comment by vfdev-5** [comment]
Hi @Nic-Ma , in ignite we provide a default score function from metric : https://pytorch.org/ignite/generated/ignite.handlers.checkpoint.Checkpoint.html#ignite.handlers.checkpoint.Checkpoint.get_default_score_fn in addition to `score_function` argument such that both cases: larger-better or lower-better are covered:
```python
from ignite.handlers import Checkpoint

best_acc_score = Checkpoint.get_default_score_fn("accuracy")
best_model_handler = Checkpoint(
    to_save, save_handler, score_name="val_accuracy", score_function=best_acc_score
)
evaluator.add_event_handler(Events.COMPLETED, best_model_handler)
```
and 
```python
from ignite.handlers import Checkpoint

neg_loss_score = Checkpoint.get_default_score_fn("loss", -1.0)
best_model_handler = Checkpoint(
    to_save, save_handler, score_name="val_neg_loss", score_function=neg_loss_score
)
evaluator.add_event_handler(Events.COMPLETED, best_model_handler)
```

**Comment by Nic-Ma** [comment]
Hi @vfdev-5 ,

Cool! I will try to update MONAI side to leverage this new ignite feature soon.
Let's keep this ticket open to track it.

Thanks.

**Comment by Nic-Ma** [comment]
Hi @vfdev-5 ,

I submitted PR https://github.com/Project-MONAI/MONAI/pull/2487 to add support for this feature request.
I think there is a minor drawback in ignite: the saved checkpoint will have a "-" in the filename:
`net_validation_loss=-0.1234.pt`
Maybe you can consider to correct it back to positive number in the filename in future ignite version?

Thanks.

**Comment by vfdev-5** [comment]
@Nic-Ma thanks ! Yes, I agree that it is better to show actual value instead of negative one. We had previously `abs(score)` such that negative values become positive but we removed that.

**Comment by Nic-Ma** [comment]
Hi @vfdev-5 , so do you plan to add back `abs(score)` logic in next ignite version?

Thanks.
